elastica/experimental/connection_contact_joint/rod_rigidbody_connection.py

```diff
@@ -0,0 +1,417 @@
+    point_system_two : numpy.ndarray
+        Describes for system two in the local coordinate system the translation from the node `index_two` (for rods)
+        or the center of mass (for rigid bodies) to the joint.
+
+
+    Examples
+    --------
+    How to connect two Cosserat rods together using a spherical joint with a gap of 0.01 m in between.
+
+    >>> simulator.connect(rod_one, rod_two, first_connect_idx=-1, second_connect_idx=0).using(
+    ...    FreeJoint,
+    ...    k=1e4,
+    ...    nu=1,
+    ...    point_system_one=np.array([0.0, 0.0, 0.005]),
+    ...    point_system_two=np.array([0.0, 0.0, -0.005]),
+    ... )
+
+    How to connect the distal end of a CosseratRod with the base of a cylinder using a spherical joint.
+
+    >>> simulator.connect(rod, cylinder, first_connect_idx=-1, second_connect_idx=0).using(
+    ...    FreeJoint,
+    ...    k=1e4,
+    ...    nu=1,
+    ...    point_system_two=np.array([0.0, 0.0, -cylinder.length / 2.]),
+    ... )
+
+    """
+
+    # pass the k and nu for the forces
+    # also the necessary systems for the joint
+    def __init__(self, k, nu, point_system_one=None, point_system_two=None):
+        """
+
+        Parameters
+        ----------
+        k: float
+           Stiffness coefficient of the joint.
+        nu: float
+           Damping coefficient of the joint.
+        point_system_one : numpy.ndarray
+            Describes for system one in the local coordinate system the translation from the node `index_one` (for rods)
+            or the center of mass (for rigid bodies) to the joint.
+            (default = np.array([0.0, 0.0, 0.0]))
+        point_system_two : numpy.ndarray
+            Describes for system two in the local coordinate system the translation from the node `index_two` (for rods)
+            or the center of mass (for rigid bodies) to the joint.
+            (default = np.array([0.0, 0.0, 0.0]))
+        """
+        self.k = k
+        self.nu = nu
+        self.point_system_one = (
+            point_system_one
+            if point_system_one is not None
+            else np.array([0.0, 0.0, 0.0])
+        )
+        self.point_system_two = (
+            point_system_two
+            if point_system_two is not None
+            else np.array([0.0, 0.0, 0.0])
+        )
+
+    def apply_forces(self, system_one, index_one, system_two, index_two):
+        """
+        Apply joint force to the connected systems.
+
+        Parameters
+        ----------
+        system_one : SystemType
+             System two of the joint connection.
+             Object of a child class of  either `CosseratRod` or `RigidBodyBase`.
+        index_one : int
+            Index of first system for joint.
+        system_two : SystemType
+            System two of the joint connection.
+            Object of a child class of  either `CosseratRod` or `RigidBodyBase`.
+        index_two : int
+            Index of second system for joint.
+
```

Comment:
if we know the joint is between a rod and a rigid body maybe like the external contact we can ask for the first body to be a rod and the second to be a rigid body? modify docstring and functionality accordingly?